firmware/thruna/FSM/Src/Io/Io_AcceleratorPedals.c

```diff
@@ -1,76 +1,132 @@
 // Denominator term for angle calculation using the cosine law
-#define PAPPS_COS_LAW_DENOMINATOR (2.0f * PAPPS_LEN_A * PAPPS_LEN_B)
-#define PAPPS_COS_LAW_COEFFICIENT (PAPPS_LEN_A * PAPPS_LEN_A + PAPPS_LEN_B * PAPPS_LEN_B) / PAPPS_COS_LAW_DENOMINATOR
-// Calculate the distance of the pedal from the voltage
-#define PAPPS_LEN_PER_VOLTAGE \
-    ((PAPPS_LENGTH_FULL_PRESSED_MM - PAPPS_LENGTH_UNPRESSED_MM) / (PAPPS_FULL_PRESSED_POT_V - PAPPS_UNPRESSED_POT_V))
-// PAPPS length offset in mm
-#define PAPPS_LEN_OFFSET_MM (227.6f)
-#define PAPPS_RAW_VOLTAGE_TO_LEN_MM(voltage) (voltage * PAPPS_LEN_PER_VOLTAGE + PAPPS_LEN_OFFSET_MM)
 
-#define MIN_PEDAL_VOLTAGE (0.5f)
-#define MAX_PEDAL_VOLTAGE (3.1f)
+#define PAPPS_COS_LAW_DENOMINATOR (2.0f * PAPPS_LEN_A * LEN_B)
+#define PAPPS_COS_LAW_COEFFICIENT ((PAPPS_LEN_A * PAPPS_LEN_A + LEN_B * LEN_B) / PAPPS_COS_LAW_DENOMINATOR)
+
+#define SAPPS_COS_LAW_DENOMINATOR (2.0f * SAPPS_LEN_A * LEN_B)
+#define SAPPS_COS_LAW_COEFFICIENT ((SAPPS_LEN_A * SAPPS_LEN_A + LEN_B * LEN_B) / SAPPS_COS_LAW_DENOMINATOR)
+
+// Macro to calculate the linear change in PAPPS length per change in voltage (we only use max because min = 0)
+
+#define APPS_VOLTAGE_PER_MM ((MAX_POT_LENGTH - MIN_POT_LENGTH) / MAX_POT_V)
+
+// linear equation to calculate current length using a rate of change from potientiometer
+// here max length here is the Yint and apps voltager per mm is the slope
 
-// NONLINEAR APPROXIMATION
-#define FULLY_UNPRESSED_POT_LENGTH_MM (212.2f)
-#define FULLY_PRESSED_POT_LENGTH_MM (165.6f)
+#define RAW_VOLTAGE_TO_LEN_MM(voltage) (MAX_POT_LENGTH - voltage * APPS_VOLTAGE_PER_MM)
 
-#define PAPPS_DEADZONE_PERCENT (10.0f)
-#define SAPPS_DEADZONE_PERCENT (8.0f)
+// calculating the unpressed and pressed PAPPS/SAPPS length based on the measured pot readings
+
+#define PAPPS_LENGTH_UNPRESSED_MM (RAW_VOLTAGE_TO_LEN_MM(PAPPS_UNPRESSED_POT_V))
+#define PAPPS_LENGTH_FULLY_PRESSED_MM (RAW_VOLTAGE_TO_LEN_MM(PAPPS_FULL_PRESSED_POT_V))
+
+#define SAPPS_LENGTH_UNPRESSED_MM (RAW_VOLTAGE_TO_LEN_MM(SAPPS_UNPRESSED_POT_V))
+#define SAPPS_LENGTH_FULLY_PRESSED_MM (RAW_VOLTAGE_TO_LEN_MM(SAPPS_FULL_PRESSED_POT_V))
+
+// OC & SC bounds
+
+#define PAPPS_MIN_V (PAPPS_UNPRESSED_POT_V - 0.5f)
+#define PAPPS_MAX_V (PAPPS_FULL_PRESSED_POT_V + 0.5f)
+
+#define SAPPS_MIN_V (SAPPS_UNPRESSED_POT_V - 0.5f)
+#define SAPPS_MAX_V (SAPPS_FULL_PRESSED_POT_V + 0.5f)
+
+#define DEAD_ZONE_PERCENT (10.3f)
+
+float calcAppsAngle(float cos_law_coefficent, float cos_law_denominator, float pot_len)
+{
+    float angle = (acosf(cos_law_coefficent - (powf(pot_len, 2) / cos_law_denominator)));
+
+    return angle;
+}
 
 float Io_AcceleratorPedals_GetPapps(void)
 {
-    // Length calculated via voltage
-    float       pedal_voltage = Io_Adc_GetChannelVoltage(ADC1_CHANNEL_2);
-    const float pot_len_mm    = PAPPS_RAW_VOLTAGE_TO_LEN_MM(pedal_voltage);
+    float pedal_voltage = Io_Adc_GetChannelVoltage(ADC1_CHANNEL_2);
+
+    // Length calculated via voltage reading from PAPPS
+
```

Comment:
nit: Extra lines like these under comments (here and elsewhere) aren't necessary